browser/src/map/Clipboard.js

```diff
@@ -872,28 +872,42 @@ L.Clipboard = L.Class.extend({
 
 	_asyncAttemptNavigatorClipboardWrite: async function(params) {
 		const command = this._unoCommandForCopyCutPaste;
-		const check_ = await this._sendCommandAndWaitForCompletion(command, params);
+		const check_ = this._sendCommandAndWaitForCompletion(command, params);
 
-		if (check_ === null)
-			return; // Either wrong command or a pending event.
-
-		// This is sent down the websocket URL which can race with the
-		// web fetch - so first step is to wait for the result of
-		// that command so we are sure the clipboard is set before
-		// fetching it.
+		// I strongly disrecommend awaiting before the clipboard.write line in the non-iOS-app path
+		// It turns out there are some rather precarious conditions for copy/paste to be allowed in Safari on mobile - and awaiting seems to tip us over into "too late to copy/paste"
+		// Deferring like this is kinda horrible - it certainly looks gross in places - but it's absolutely necessary to avoid errors on the clipboard.write line
+		// I don't like it either :). If you change this make sure to thoroughly test cross-browser and cross-device!
 
 		if (window.ThisIsTheiOSApp) {
+			// This is sent down the fakewebsocket which can race with the
+			// native message - so first step is to wait for the result of
+			// that command so we are sure the clipboard is set before
+			// fetching it.
+			if (await check_ === null)
+				return; // Either wrong command or a pending event.
+
 			await window.webkit.messageHandlers.clipboard.postMessage(`write`);
 		} else {
 			const url = this.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
 
-			var result = await fetch(url);
-			var text = await result.text();
+			const text = (async () => {
```

Comment:
OK, so we used to do clipboard.navigator.write() outside the security context and now we put a promise to the clipboard, inside the security context. I think that's what my initial async clipboard code did -- and then somehow this was broken & we failed to catch that during code review.

Also, you maintain the invariant that the dispatch of the .uno:Copy is async, so you have to wait for it and only then you can fetch the URL. At least the normal desktop browser copy&paste new version looks correct to me.